client/src/test/java/com/sinch/sdk/e2e/domains/conversation/AppsSteps.java

```diff
@@ -0,0 +1,140 @@
+                    .setClaimedIdentity(identity)
+                    .setToken(token)
+                    .build())
+            .build();
+
+    AppCreateRequest request =
+        AppCreateRequest.builder()
+            .setChannelCredentials(Arrays.asList(credentials))
+            .setDisplayName(displayName)
+            .build();
+
+    createResponse = service.create(request);
+  }
+
+  @When("^I send a request to list all the apps$")
+  public void list() {
+
+    listResponse = service.list();
+  }
+
+  @When("^I send a request to retrieve an app$")
+  public void get() {
+
+    getResponse = service.get("foo");
+  }
+
+  @When("^I send a request to update an app$")
+  public void update() {
+
+    AppUpdateRequest request =
+        AppUpdateRequest.builder().setDisplayName(" a new display name").build();
+    updateResponse = service.update("foo", request);
+  }
+
+  @When("^I send a request to delete an app$")
+  public void delete() {
+
+    service.delete("foo");
+    deletePassed = true;
+  }
+
+  @Then("the conversation app is created")
+  public void createResult() {
+    checkExpectedAppResponseCommonFields(createResponse);
+    Assertions.assertEquals(
+        createResponse.getChannelCredentials().get(0).getState().getStatus(),
+        ChannelIntegrationStatus.PENDING);
+  }
+
+  @Then("the apps list contains {int} apps")
+  public void listResult(int size) {
+
+    Assertions.assertEquals(listResponse.size(), size);
+    AppResponse item = listResponse.stream().findFirst().orElse(null);
+
+    checkExpectedAppResponseCommonFields(item);
+    Assertions.assertEquals(item.getDisplayName(), "E2E Conversation App");
+    Assertions.assertEquals(
+        item.getChannelCredentials().get(0).getState().getStatus(),
+        ChannelIntegrationStatus.ACTIVE);
+    item = listResponse.stream().reduce((first, second) -> second).orElse(null);
+    Assertions.assertEquals(item.getId(), "01W4FFL35P4NC4K35CONVAPP002");
+  }
+
+  @Then("the response contains the app details")
+  public void getResult() {
+    checkExpectedAppResponseCommonFields(getResponse);
+    Assertions.assertEquals(getResponse.getDisplayName(), "E2E Conversation App");
+    Assertions.assertEquals(
+        getResponse.getChannelCredentials().get(0).getState().getStatus(),
+        ChannelIntegrationStatus.ACTIVE);
+  }
+
+  @Then("the response contains the app details with updated properties")
+  public void updateResult() {
+    checkExpectedAppResponseCommonFields(updateResponse);
+    Assertions.assertEquals(updateResponse.getDisplayName(), "Updated name");
+    Assertions.assertEquals(
+        updateResponse.getChannelCredentials().get(0).getState().getStatus(),
+        ChannelIntegrationStatus.ACTIVE);
+  }
+
+  @Then("the delete app response contains no data")
+  public void deleteResult() {
+
+    Assertions.assertTrue(deletePassed);
+  }
+
+  void checkExpectedAppResponseCommonFields(AppResponse appResponse) {
```

Comment:
This is a nice idea to factorize the check on common fields. But some tests can break if we change some data in the mockserver expectations. I've tried to create some consistency when I wrote them. Let's agree together to keep it like this!

Reply:
According to our discussion: we agree to update tests accordingly in case of future change